front-end/js/jump.js

```diff
@@ -362,6 +390,147 @@ function registerJumps(viewId, svg, layerId) {
         // make cursor a hand when hovering over this shape
         d3.select(this).style("cursor", "zoom-in");
 
+        // disable context menu on kyrix objects since we use right clicks for update
+        d3.select(this).on("contextmenu", function(d) {
+            d3.event.preventDefault();
+        });
+
+        // register right click listener -- for update popover
+        d3.select(this).on("auxclick", function (d) {
+            // only register listener logic if right click
+            if (d3.event.button != 2) {
+                return;
+            }
+
+            // stop the right click event from propagating up
+            d3.event.preventDefault();
+            d3.event.stopPropagation();
+            
+            // gvd - data for current view, current canvas, transform, etc.
+            let canvasId = gvd.curCanvasId;
+            let queryText = gvd.curCanvas.layers[layerId].transform.query;
+
+            // use regex to extract db column names from user-defined transform
+            [_, queryText] = queryText.split("select");
+            [queryText, tableName] = queryText.split("from");
+            tableName = tableName.replace(/[\s;]+/g, "").trim();
+            queryText = queryText.replace(/\s+/g, "").trim();
```

Comment:
Please comment that this is a fragile hack for now, and e.g. the substring "select" or "from" just appear once in the SQL statement e.g. cannot appear in SQL comments

Why do you need to replace whitespace and then trim() ?